### PR TITLE
Burger menu & Profile page

### PR DESCRIPTION
**Pull request by shansty**

### CODE REVIEW

File: src/components/Register.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ const Register = () => {
         }
         try {
             const response = await axios.post(REGISTER_URL,
-                { fullName: user, email: email, password: pwd }
+                { userName: user, email: email, password: pwd }
```
Comment (busha98) [inline src/components/Register.js:69]:
```
{ email, userName: user, password: pwd }
```

File: src/components/Register.js [hunk 2]
```diff
@@ -218,7 +218,7 @@ const Register = () => {
                         Must match the first password input field.
                     </p>
 
-                    <button type="submit" disabled={!validEmail || !validName || !validPwd || !validMatch ? true : false}> Sign Up </button>
+                    <button className="auth_button" type="submit" disabled={!validEmail || !validName || !validPwd || !validMatch ? true : false}> Sign Up </button>
```
Comment (busha98) [inline src/components/Register.js:221]:
```suggestion
                    <button className="auth_button" type="submit" disabled={!validEmail || !validName || !validPwd || !validMatch}> Sign Up </button>
```

File: src/utils.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import Login from "./components/Login";
+
+function getIDFromToken() {
+    let savedToken = localStorage.getItem("token");
```
Comment (busha98) [inline src/utils.js:4]:
```suggestion
    const token= localStorage.getItem("token");
```

File: src/utils.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    if(!savedToken) {
+        return(
+            <Login />
```
Comment (busha98) [inline src/utils.js:7]: redirect to login

File: src/components/profile/Profile.js [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        setUser({
+            userName: userName,
+            email: email,
+            phoneNumber: phoneNumber,
+            shopOwner: shopOwner
+        })
+    }, [userName, email, phoneNumber, shopOwner]);
+
+    useEffect(() => {
+        getUser()
```
Comment (busha98) [inline src/components/profile/Profile.js:28]: call a function before the declaration

File: src/components/profile/Profile.js [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    }, []);
+
+    const getUser = async (e) => {
+        try {
+            const response = await axios.get(PROFILE_URL, localStorage.getItem("token"),
```
Comment (busha98) [inline src/components/profile/Profile.js:33]: ?

Comment (busha98) [reply src/components/profile/Profile.js:33]: add token to Authorization header
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization
https://habr.com/ru/companies/doubletapp/articles/764424/

File: src/components/profile/Profile.js [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+    const handleEdit = (e) => {
+        e.preventDefault();
+        setAllowEdit(true);
+    };
+
+    const handleSubmit = async (e) => {
+        setAllowEdit(false);
+        try {
+            await axios.patch(PROFILE_URL, {user},  
+                {headers: {'Access-Control-Allow-Origin': '*', 'Content-Type': 'application/json'}});
```
Comment (busha98) [inline src/components/profile/Profile.js:57]: add Authorization header

File: src/components/Login.js [hunk 8]
```diff
@@ -22,18 +22,20 @@ const Login = () => {
...
     useEffect(() => {
         setErrMsq("");
-    }, [user, pwd])
+    }, [email, pwd])
 
     const handleSubmit = async (e) => {
         e.preventDefault();
         try {
-            const response = await axios.post(LOGIN_URL, {email:user, password:pwd},  
+            const response = await axios.post(LOGIN_URL, {email:email, password:pwd},
```
[the remark this answers is not in the document]
Comment (shansty) [reply src/components/Login.js:30]: +

[the remark this answers is not in the document]
Comment (shansty) [reply src/components/Login.js:30]: fixed

File: src/components/Register.js (hunk 1 above)
Comment (shansty) [reply src/components/Register.js:69]: fixed

File: src/utils.js (hunk 3 above)
Comment (shansty) [reply src/utils.js:4]: fixed

File: src/components/profile/Profile.js (hunk 6 above)
Comment (shansty) [reply src/components/profile/Profile.js:33]: fixed

File: src/components/profile/Profile.js (hunk 7 above)
Comment (shansty) [reply src/components/profile/Profile.js:57]: fixed

File: src/utils.js [hunk 9]
```diff
@@ -0,0 +1,13 @@
+function getIDFromToken() {
+    const token = localStorage.getItem("token");
+    if(!token) {
+        window.location.replace('/login');
```
Comment (busha98) [inline src/utils.js:4]: https://www.youtube.com/watch?v=i0rbmzyaVaM

File: src/components/profile/Profile.js [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+
+    useEffect(() => {
+        setUser(user => ({
+          ...user,
+          phoneNumber: user.phoneNumber,
+          shopOwner: user.shopOwner,
+          userName: user.userName,
+          email: user.email
+        }));
+      }, [user.shopOwner, user.email, user.userName, user.phoneNumber]);
```
Comment (busha98) [inline src/components/profile/Profile.js:71]:
```suggestion
      }, [user]);
```

**Review by busha98 — APPROVED** [review]

**Pull request merged**